sql/src/main/java/io/crate/planner/optimizer/LoadedRules.java

```diff
@@ -0,0 +1,126 @@
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.optimizer;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CaseFormat;
+import io.crate.common.collections.Lists2;
+import io.crate.metadata.settings.session.SessionSetting;
+import io.crate.metadata.settings.session.SessionSettingProvider;
+import io.crate.planner.operators.RewriteInsertFromSubQueryToInsertFromValues;
+import io.crate.planner.optimizer.rule.DeduplicateOrder;
+import io.crate.planner.optimizer.rule.MergeAggregateAndCollectToCount;
+import io.crate.planner.optimizer.rule.MergeFilterAndCollect;
+import io.crate.planner.optimizer.rule.MergeFilters;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathFetchOrEval;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathGroupBy;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathHashJoin;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathNestedLoop;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathOrder;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathProjectSet;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathRename;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathUnion;
+import io.crate.planner.optimizer.rule.MoveFilterBeneathWindowAgg;
+import io.crate.planner.optimizer.rule.MoveOrderBeneathFetchOrEval;
+import io.crate.planner.optimizer.rule.MoveOrderBeneathNestedLoop;
+import io.crate.planner.optimizer.rule.MoveOrderBeneathRename;
+import io.crate.planner.optimizer.rule.MoveOrderBeneathUnion;
+import io.crate.planner.optimizer.rule.RemoveRedundantFetchOrEval;
+import io.crate.planner.optimizer.rule.RewriteCollectToGet;
+import io.crate.planner.optimizer.rule.RewriteFilterOnOuterJoinToInnerJoin;
+import io.crate.planner.optimizer.rule.RewriteGroupByKeysLimitToTopNDistinct;
+import io.crate.planner.optimizer.rule.RewriteToQueryThenFetch;
+import io.crate.types.DataTypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+public class LoadedRules implements SessionSettingProvider {
+
+    private static final String OPTIMIZER_RULE = "optimizer_";
+
+    private static final List<Rule<?>> RULES = List.of(
```

Comment:
I wonder if these should actually be instance local, and if we should have the `Optimizer` and `SessionRegistry` use the same instance. Otherwise in our tests where we simulate multiple nodes  within the same JVM we will have session settings shared **across** nodes. So our tests won't reflect real-world scenarios.